libsolidity/interface/CompilerStack.cpp

```diff
@@ -114,9 +115,10 @@ void CompilerStack::setSource(string const& _sourceCode)
 bool CompilerStack::parse()
 {
 	//reset
+	if(m_stackState < SourcesSet)
```

Comment:
Probably it would be safer checking for equality here, e.g. disallowing running parse again.